### ISSUE TITLE
function for the api

### ISSUE THREAD

**Issue by endes123321** [cuberite/cuberite#2373]
Hello, I recently discovered cuberite and I start making plugins, but I have been limited somewhat by the api and I would like to propose these functions and classes for the api:

* a function to send a scoreboard just a player and not all players
* a function for riding a entity
* a function for send titles
* a class to interact directly with the protocol
* a class to interact with an entity ai
* a class to add NBT tags to block and entities

**Comment by worktycho** [comment]
what do you mean by interact with the protocol? MCServer is written to support multiple protocol versions, so low level protocol changes can get complicated.

**Comment by NiLSPACE** [comment]
A interface like [this](https://github.com/cuberite/cuberite/pull/1826#issuecomment-84707564) could be used for this. It would allow developers to send other packets then the broadcastXYZ packets, or send the same type of packets, but with information that can't be send when using the broadcast functions.

**Comment by worktycho** [comment]
I don't think that exposing protocol stuff is bad, just that we need to be careful.

**Comment by krisk557** [comment]
The scoreboard stuff would be really useful.
Currently it's the only thing that prevents me and my team to completely drop craftbukkit and migrate to cuberite. :(

**Comment by SamJBarney** [comment]
The scoreboard per player was something I was going to ask be implemented :P
I personally think it is a very good idea

**Comment by LO1ZB** [comment]
@SamJBarney that feature is used by almost every minigame server I know.

**Comment by endes123321** [comment]
@krisk557 Scoreboard per player and editing NBT tags are very important in my opinion.
@worktycho  can interact with the protocol would be an advantage for Cubelite

**Comment by SphinxC0re** [comment]
I updated your comment so we now have a task list
